### PR TITLE
Prometheus - complete

### PR DESCRIPTION
**Pull request by tenisakb**

### CODE REVIEW

**Comment by tenisakb** [comment]
The previous PR was failed but it was error on travis side

File: actions/transaction_brokernodes.go [hunk 1]
```diff
@@ -100,6 +104,9 @@ func (usr *TransactionBrokernodeResource) Create(c buffalo.Context) error {
 }
 
 func (usr *TransactionBrokernodeResource) Update(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/transaction_brokernodes.go]: Might should change this to HistogramTransactionBrokernodeResourceUpdate?

File: actions/transaction_genesis_hashes.go [hunk 2]
```diff
@@ -46,6 +46,9 @@ type transactionGenesisHashUpdateRes struct {
 
 // Creates a transaction.
 func (usr *TransactionGenesisHashResource) Create(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/transaction_genesis_hashes.go]: HistogramTransactionGenesisHashResourceCreate?

File: actions/transaction_genesis_hashes.go [hunk 3]
```diff
@@ -104,6 +107,9 @@ func (usr *TransactionGenesisHashResource) Create(c buffalo.Context) error {
 }
 
 func (usr *TransactionGenesisHashResource) Update(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/transaction_genesis_hashes.go]: HistogramTransactionGenesisHashResourceUpdate?

File: prometheus.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    static_configs:
+      - targets: ['localhost:9090']
+
+  - job_name: 'brokernode'
+
+    # metrics_path defaults to '/metrics'
+    # scheme defaults to 'http'.
+    static_configs:
+      - targets: ['localhost:3000']
```
Comment (rfornea) [inline prometheus.yml]: Do you think we will be able to see this from the broker's IP addresses?  Like if I go to (broker_ip):9090 will I be able to see the Prometheus stuff?

File: actions/treasures.go [hunk 5]
```diff
@@ -25,11 +25,13 @@ type treasureRes struct {
 
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/treasures.go]: Here as well

File: actions/upload_sessions.go [hunk 6]
```diff
@@ -194,6 +196,8 @@ func (usr *UploadSessionResource) Create(c buffalo.Context) error {
 
 // Update uploads a chunk associated with an upload session.
 func (usr *UploadSessionResource) Update(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/upload_sessions.go]: Here

File: actions/upload_sessions.go [hunk 7]
```diff
@@ -332,6 +336,9 @@ func (usr *UploadSessionResource) Update(c buffalo.Context) error {
 
 // CreateBeta creates an upload session on the beta broker.
 func (usr *UploadSessionResource) CreateBeta(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/upload_sessions.go]: Here

File: actions/upload_sessions.go [hunk 8]
```diff
@@ -407,6 +414,9 @@ func (usr *UploadSessionResource) CreateBeta(c buffalo.Context) error {
 }
 
 func (usr *UploadSessionResource) GetPaymentStatus(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/upload_sessions.go]: Here

File: actions/webnodes.go [hunk 9]
```diff
@@ -22,6 +22,9 @@ type webnodeCreateRes struct {
 
 // Creates a webnode.
 func (usr *WebnodeResource) Create(c buffalo.Context) error {
+	start := PrometheusWrapper.TimeNow()
+	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramTransactionBrokernodeResourceCreate, start)
```
Comment (rfornea) [inline actions/webnodes.go]: Here

**Pull request merged**